lib/rspec/mocks/proxy.rb

```diff
@@ -35,8 +35,7 @@ def initialize(object, order_group, options={})
 
       # @private
       def ensure_can_be_proxied!(object)
-        return unless object.is_a?(Symbol) || object.frozen?
-        return if object.nil?
+        return unless object.is_a?(Symbol) || (object.is_a?(String) && object.frozen?)
```

Comment:
This check is wrong, we still need to check other types of objects are frozen. If you want to improve this check to actually check the real object is frozen please do, (prehaps checking that the frozen method is kernels method) but as it is I don't want to only warn for frozen strings.

Reply:
It's actually simply redundant, the new exception handling takes care of all objects regardless of this check. I had mentioned this previously in https://github.com/rspec/rspec-mocks/pull/1401#discussion_r810679191 and @pirj advised minimizing the changes, but I would be happy to remove this method entirely to avoid this confusion.